instrumentation/src/test/java/io/opentelemetry/android/emitter/EventEmitterWithDataTest.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.emitter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Test;
+
+public class EventEmitterWithDataTest {
+
+    @Test
+    public void testConvertRichDataObjectIntoAttributes() {
+        Map<String, Object> data = new HashMap<>();
+        data.put("stringKey", "key");
```

Comment:
Can we use an actual android crash data here?

Reply:
Done, preserved the original test fields since they include double, boolean etc for other possible future cases